component/org.wso2.carbon.identity.casque.authenticator/src/main/java/org/wso2/carbon/identity/casque/authenticator/authenticator/AuthPages.java

```diff
@@ -0,0 +1,107 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.wso2.carbon.identity.casque.authenticator.authenticator;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.casque.authenticator.authenticator.radius.RadiusPacket;
+import org.wso2.carbon.identity.casque.authenticator.constants.CasqueAuthenticatorConstants;
+import org.wso2.carbon.identity.casque.authenticator.exception.CasqueException;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Serializable;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * This Class Load the ChallengPage and QR player
+ */
+class AuthPages implements Serializable {
+
+    private static final long serialVersionUID = 4341535155455223655L;
+    private static final Log log = LogFactory.getLog(RadiusPacket.class);
+
+    private void returnHtmlResponse(HttpServletResponse response, String data) throws IOException {
+
+        response.setContentType(CasqueAuthenticatorConstants.CONTENT_TYPE);
+        returnResponse(response, data);
+    }
+
+    /**
+     * Return the response
+     *
+     * @param response Http servlet response
+     * @param data     get data from the response
+     * @throws IOException :
+     */
+    private void returnResponse(HttpServletResponse response, String data) throws IOException {
+
+        response.addHeader(CasqueAuthenticatorConstants.CACHE_CONTROL, "no-cache");
+        response.addHeader(CasqueAuthenticatorConstants.PRAGMA, "no-cache");
+        response.addHeader(CasqueAuthenticatorConstants.EXPIRES, "0");
+        response.setContentLength(data.length());
+        response.getOutputStream().print(data);
+    }
+
+    /**
+     * load the challengPage
+     *
+     * @param response       http servelet request
+     * @param sessionDataKey ,
+     * @param challenge      challenge for QR_Player
+     * @throws CasqueException
+     */
+    void challengePage(HttpServletResponse response, String sessionDataKey, String challenge) throws CasqueException {
+
+        try {
+            String resource = loadResource(CasqueAuthenticatorConstants.QR_PLAYER);
+            if (resource != null) {
+                resource = resource.replace(CasqueAuthenticatorConstants.CASQUE_CHALLENGE, challenge);
+                resource = resource.replace(CasqueAuthenticatorConstants.SESSION_DATA_KEY, sessionDataKey);
+                returnHtmlResponse(response, resource);
+            } else {
+                log.error("QR player resources are not available");
```

Comment:
If the QR_PLAYER resource is not there, then it is enough to print the log and continue the flow? Don't we need to throw an exception?

Reply:
fixed